src/List/Selection.elm

```diff
@@ -115,3 +151,43 @@ map fn (Selection selected items) =
     Selection
         (Maybe.map fn selected)
         (List.map fn items)
+
+
+{-| Apply a function to only the selected item, if any item is selected.
+-}
+mapSelected : (a -> a) -> Selection a -> Selection a
```

Comment:
I wonder if having a function like `format` would be better.

```elm
format : { unselected : a -> b, selected: a -> b } -> Selection a -> Selection b
```
I proposed the same for listzipper.
This allows you to just map all the unselected as well.

Reply:
You could also potentially remove `toListWithSelected` with this.

```elm
selection
|> format { unselected = (,) False, selected = (,) True }
|> toList
```

I like having less api. We could still create extra module with convenient funcitons.

Also there might be someone how doesn't want `bool`, but their type.

```
selection
|> format { unselected = (,) NotStarred, selected = (,) Starred }
|> toList
```

Reply:
I like this! I think `format` is unintuitive given `map` though, so I'm gonna still call it `mapSelected`, because it's "map special casing selected".